Save the sites in PouchDB instead of LocalStorage

danielmilner opened the pull request:
Quite a few changes needed to be made. It's not using SQL Lite yet, but from what I understand, it will automatically pick it up if the plugin is detected.

danielmilner committed:
Moved the sites cache over to PouchDB.

danielmilner committed:
Changed the siteID to not rely on localStorage.

danielmilner committed:
Added back in the logic to remove the page, post, and comment cache when a site is deleted.

shprink commented on www/js/controllers.js:
```diff
@@ -24,22 +24,35 @@ angular.module('wpApp.controllers', [])
-  }
-
+  
+  $scope.loadSites = function() {
```
What's the purpose of binding this function to the scope? Are you gonna use it in a view somewhere?

danielmilner replied:
I guess it's not necessary. That's just how I'm used to doing it.

shprink replied:
You can use named functions instead:

```
function loadSites(){}
```

scottopolis commented on www/js/controllers.js:
```diff
@@ -73,12 +86,14 @@ angular.module('wpApp.controllers', [])
-        var siteID = $rootScope.increment();
-        var site = { id: siteID, title: response.data.name, description: response.data.description, url: siteURL, username: u.username, password: u.password };
-        $scope.sites.push( site );
+        var site = { _id: new Date().toISOString(), title: response.data.name, description: response.data.description, url: siteURL, username: u.username, password: u.password };
```
Hmm, site ID as date is a pretty long string, is that necessary?

danielmilner replied:
It was recommended in the PouchDB documentation http://pouchdb.com/getting-started.html

If we want to use something else, that's totally fine. I just didn't want to get it from localStorage anymore just in case that got wiped.

scottopolis replied:
Ya it makes sense, if the increment got wiped that would be a problem.

scottopolis commented on www/js/controllers.js:
```diff
@@ -212,7 +234,7 @@ angular.module('wpApp.controllers', [])
     $timeout(function() {
 
-      DataLoader.get( dataURL + '?page=' + pg + '&' + $rootScope.callback ).then(function(response) {
+      DataLoader.get( $scope.dataURL + '?page=' + pg + '&' + $rootScope.callback ).then(function(response) {
```
don't need $scope here, dataURL is not used in a view

danielmilner committed:
Removed the `loadSites()` function from the `$scope`.
Removed the `dataURL` variable from the `$scope`.
Fixed an issue with saving a site.
Added back in the CacheFactory check so that page caching will work on new installs.

danielmilner committed:
I had missed one of the `dataURL` variables.

danielmilner committed:
Extended the timeout for the `loadMore` function so that it runs after the `dataURL` has had time to populate from the database.

danielmilner committed:
Merge branch 'master' of https://github.com/wp-app/wp-app

Conflicts:
	www/js/controllers.js

danielmilner committed:
Merged in scottopolis' commit from wp-app:master

danielmilner commented:
@scottopolis I merged in commit # 6525cbcc6f43757d1d81975244c3475721eb8d0c

danielmilner commented:
I was doing some more research on PouchDB last night. I think that I can clean up some of the code used to access the database.

danielmilner committed:
Removed the database init.

danielmilner committed:
Added a new database service for interacting with PouchDB.

danielmilner committed:
Made the controllers work with the new SitesDB service.
